src/dbnode/client/session.go

```diff
@@ -182,26 +183,28 @@ type shardMetricsKey struct {
 
 type sessionMetrics struct {
 	sync.RWMutex
-	writeSuccess                         tally.Counter
-	writeErrorsBadRequest                tally.Counter
-	writeErrorsInternalError             tally.Counter
-	writeLatencyHistogram                tally.Histogram
-	writeNodesRespondingErrors           []tally.Counter
-	writeNodesRespondingBadRequestErrors []tally.Counter
-	fetchSuccess                         tally.Counter
-	fetchErrorsBadRequest                tally.Counter
-	fetchErrorsInternalError             tally.Counter
-	fetchLatencyHistogram                tally.Histogram
-	fetchNodesRespondingErrors           []tally.Counter
-	fetchNodesRespondingBadRequestErrors []tally.Counter
-	topologyUpdatedSuccess               tally.Counter
-	topologyUpdatedError                 tally.Counter
-	streamFromPeersMetrics               map[shardMetricsKey]streamFromPeersMetrics
+	writeSuccess                                     tally.Counter
+	writeSuccessForCountLeavingAndInitializingAsPair tally.Counter
+	writeErrorsBadRequest                            tally.Counter
+	writeErrorsInternalError                         tally.Counter
+	writeLatencyHistogram                            tally.Histogram
+	writeNodesRespondingErrors                       []tally.Counter
+	writeNodesRespondingBadRequestErrors             []tally.Counter
+	fetchSuccess                                     tally.Counter
+	fetchErrorsBadRequest                            tally.Counter
+	fetchErrorsInternalError                         tally.Counter
+	fetchLatencyHistogram                            tally.Histogram
+	fetchNodesRespondingErrors                       []tally.Counter
+	fetchNodesRespondingBadRequestErrors             []tally.Counter
+	topologyUpdatedSuccess                           tally.Counter
+	topologyUpdatedError                             tally.Counter
+	streamFromPeersMetrics                           map[shardMetricsKey]streamFromPeersMetrics
 }
 
 func newSessionMetrics(scope tally.Scope) sessionMetrics {
 	return sessionMetrics{
 		writeSuccess: scope.Counter("write.success"),
+		writeSuccessForCountLeavingAndInitializingAsPair: scope.Counter("write.success.leaving_and_initializing_as_pair"),
```

Comment:
nit: you'd be better off adding a metric w/ the same name as writeSuccess, and a new tag to distinguish this sub-case.

it'll make your queries a lot nicer when you use this metric